Add: room description matching to generic mapper

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Add opt-in support for matching rooms based on their descriptions in addition to names and exits. This helps distinguish rooms that share the same name and exits but have different descriptions.

#### Motivation for adding to Mudlet
Address #8004
#### Other info (issues closed, discussion etc)
Enable with `map config use_description_matching true`

claude committed:
Feature: Add room description matching to generic mapper (closes #8004)

Add opt-in support for matching rooms based on their descriptions in
addition to names and exits. This helps distinguish rooms that share
the same name and exits but have different descriptions.

Changes:
- Add `use_description_matching` config option (default: false)
- Extract room description from captured lines in name_search()
- Store descriptions in room user data when creating new rooms
- Add description comparison in check_room() for room matching
- Bump version to 2.1.8

vadi2 committed:
Add documentation and debug output for use_description_matching

vadi2 committed:
Enhancement: Show and store room descriptions in generic mapper

- Display stored room description in 'rl' (room look) alias output
- Store description for existing rooms when matched by name+exits
  (when use_description_matching is enabled)

vadi2 commented:
@zookaongit if you want to have a look

ZookaOnGit reviewed (approved):
Very occasionally it will grab the room name as well as the description which will require a remap.  But this is a vast improvement with my testing.  

<img width="931" height="371" alt="Image" src="https://github.com/user-attachments/assets/c1dd5188-8099-4827-bc87-447116b22116" />

Don't forget to package it up, otherwise it doesn't get added to the build.

vadi2 committed:
Packaging: Rebuild generic_mapper.mpackage with description matching

Rebuilds the mpackage to include the room description matching changes
from the previous commits. Updates package version to 2.1.8.

vadi2 committed:
Fix: Room title occasionally captured in description

Track the actual line index where last_line was found instead of
incorrectly using line_count - 1. This fixes cases where empty lines
between room title and prompt caused the description to include the
room title.

vadi2 commented:
Perfect thanks for the feedback. Descriptions shouldn't capture room names anymore, and the mpackage was rebuilt.
